### PR TITLE
GET: api/topics and EH

### PR DESCRIPTION
**Pull request by rjg55**
responds with correct array, tests written for status 200 and error handling of an unknown endpoint (404)

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -1,9 +1,21 @@
...
+} = require("./controllers/news.controllers");
 
 app.use(express.json());
 
+app.get("/api/topics", getTopics);
+
+// This will handle all undefined endpoints
+
+app.get("*", (req, res) => {
+  console.log("unknown endpoint");
```
Comment (k-huggs) [inline app.js]: delete console.log

File: app.js [hunk 2]
```diff
@@ -1,9 +1,21 @@
...
+  handleUnknownEndpoints,
+} = require("./controllers/news.controllers");
 
 app.use(express.json());
 
+app.get("/api/topics", getTopics);
+
+// This will handle all undefined endpoints
+
+app.get("*", (req, res) => {
```
Comment (k-huggs) [inline app.js]: you don't want to do a get request for your errors

File: app.js (hunk 1 above)
Comment (rjg55) [reply app.js]: Sorry this code shoudln't have been pushed, I forgot to save before pushing D=

File: controllers/news.controllers.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+const { selectTopics } = require("../models/news.models");
+
+exports.getTopics = (req, res, next) => {
+  selectTopics().then((response) => {
+    res.status(200).send({ topics: response });
+  });
+};
+
+exports.handleUnknownEndpoints = (req, res, next) => {
+  console.log("im an unknown endpoint controller");
```
Comment (k-huggs) [inline controllers/news.controllers.js]: delete console.log

File: controllers/news.controllers.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+const { selectTopics } = require("../models/news.models");
+
+exports.getTopics = (req, res, next) => {
+  selectTopics().then((response) => {
+    res.status(200).send({ topics: response });
+  });
+};
+
+exports.handleUnknownEndpoints = (req, res, next) => {
```
Comment (k-huggs) [inline controllers/news.controllers.js]: do you want to handle your errors in the same file as your other controller functions?

**Comment by k-huggs** [comment]
couple amendments then merge

**Pull request merged**